### PR TITLE
aws(redshiftserverless): add Redshift Serverless follow-up imports

### PR DESCRIPTION
**Pull request by chenrui333**
## Summary

- add Redshift Serverless import discovery for usage limits, endpoint access, custom domain associations, snapshots, and snapshot resource policies
- seed import IDs as provider-verified names, IDs, ARNs, or workgroup/domain composites
- keep follow-up loaders optional so AccessDenied/ResourceNotFound does not regress namespace/workgroup imports
- update docs/aws.md for supported Redshift Serverless resources
- add unit tests for Redshift Serverless helper behavior, status guards, optional loader errors, and policy heredoc wrapping

## References

- #338

## Validation

```bash
go test ./providers/aws -run 'Test.*RedshiftServerless|Test.*Redshiftserverless|Test.*redshiftserverless'
go test ./providers/aws/... ./terraformutils/...
go run ./tools/aws-gap-inventory \
  -docs docs/aws.md \
  -aws-dir providers/aws \
  -skip-list providers/aws/unsupported_resources.json \
  -format markdown
git diff --check
```

## Notes

Skipped/deferred resources:
- aws_redshiftserverless_scheduled_action: not present in the current Terraform AWS provider resource docs/source inspected for this PR

---
## Summary by cubic
Adds follow-up import support for Redshift Serverless resources: `aws_redshiftserverless_usage_limit`, `aws_redshiftserverless_endpoint_access`, `aws_redshiftserverless_custom_domain_association`, `aws_redshiftserverless_snapshot`, and `aws_redshiftserverless_resource_policy`. Loaders are optional so AccessDenied/ResourceNotFound won’t block namespace/workgroup imports.

- **New Features**
  - Discovers and seeds import IDs using provider-verified names/IDs/ARNs; custom domains use a workgroup,domain composite.
  - Adds status guards (snapshots only when available; endpoint access not in creating/deleting/failed).
  - Fetches snapshot resource policies and wraps the `policy` field in a heredoc during post-convert to avoid interpolation issues.
  - Updates `docs/aws.md` for supported resources and adds unit tests for helpers, guards, optional loader behavior, and policy wrapping.

<sup>Written for commit 1c16effece66e0f649bc8d6e6bba0c6790e47235. Summary will update on new commits.</sup>

## Release Notes

* **New Features**
  * Added support for discovering and importing additional Redshift Serverless resource types: snapshots, usage limits, endpoint access, custom domain associations, and resource policies.
  * Automatic formatting support for resource policy configurations.

* **Documentation**
  * Extended Redshift Serverless documentation to include all newly supported resource types.

* **Tests**
  * Comprehensive test coverage added for new Redshift Serverless functionality, including edge cases and error handling.

[Review Change Stack](https://app.coderabbit.ai/change-stack/chenrui333/terraformer/pull/425)

### CODE REVIEW

File: providers/aws/redshiftserverless.go [hunk 1]
```diff
@@ -149,6 +442,38 @@ func redshiftServerlessWorkgroupImportable(workgroup redshiftserverlesstypes.Wor
...
+		StringValue(endpoint.WorkgroupName) != "" &&
+		len(endpoint.SubnetIds) > 0 &&
+		redshiftServerlessEndpointAccessStatusImportable(StringValue(endpoint.EndpointStatus))
+}
+
+func redshiftServerlessCustomDomainAssociationImportable(association redshiftserverlesstypes.Association) bool {
+	return redshiftServerlessCustomDomainAssociationImportID(association) != "" &&
+		StringValue(association.CustomDomainCertificateArn) != "" &&
+		association.CustomDomainCertificateExpiryTime != nil
+}
```
[the remark this answers is not in the document]
Comment (chenrui333) [reply providers/aws/redshiftserverless.go:470]: I checked this against the current Terraform AWS provider read path. Although `custom_domain_certificate_expiry_time` is computed rather than a config argument, the provider helper for `aws_redshiftserverless_custom_domain_association` currently returns an empty-result error when `GetCustomDomainAssociation` returns nil expiry:

`findCustomDomainAssociationByTwoPartKey` checks `output == nil || output.CustomDomainCertificateExpiryTime == nil`.

Since Terraformer needs to emit only resources that can refresh after import, keeping this guard avoids generating an import that the provider read path may reject. Leaving this as-is and resolving.
